libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Security/SFUserAccountManager.m

```diff
@@ -155,6 +181,9 @@ - (BOOL)isIdentityProvider {
 }
 
 - (void)setIsIdentityProvider:(BOOL)isIdentityProvider{
+    if (isIdentityProvider) {
```

Comment:
You need an `else` block too that calls `[SFSDKAppFeatureMarkers removeAppFeature:kSFIDPAppFeatureIDPLogin];`, since the flag can be toggled the other way and we'd have to remove the existing app marker then. This can happen when a new value is pushed down through MDM that enables/disables this feature.